### PR TITLE
use browser-detector instead of whichbrowser library

### PR DESCRIPTION
**Pull request by hutnikau**
We use `browser-detector` library in taoClientRestrict instead of `whichbrowser` which is more accurate.

Also because of different libraries used in taoClientRestrict and taoClientDiagnostic sometimes one extension says that user browser/OS is compatible and another says that it is not compatible.

Depends on https://github.com/oat-sa/extension-tao-client-restrict/pull/13

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -24,10 +24,11 @@
     'label' => 'Browser and OS diagnostic tool',
     'description' => 'Check compatibility of the os and browser of a client',
     'license' => 'GPL-2.0',
-    'version' => '1.8.0',
+    'version' => '1.9.0',
     'author' => 'Open Assessment Technologies SA',
     'requires' => array(
         'tao' => '>=4.0.0',
+        'taoClientRestrict' => '>=1.0.7',
```
Comment (antoinerobin) [inline manifest.php]: Unfortunately we can't afford to have this dependency. Most of instance with client-diag won't have client restrict installed

Comment (hutnikau) [inline manifest.php]: What do you think if I move `browser-detector` to `taoClientDiagnostic` and inverse dependency (make `taoClientRestrict` dependent on `taoClientDiagnostic`?

Comment (antoinerobin) [inline manifest.php]: I think this is a better solution, yes.

**Comment by hutnikau** [comment]
@antoinerobin I made necessary changes. Please review it again.

**Pull request merged**